Source/Scene/ModelExperimental/ModelExperimental.js

```diff
@@ -0,0 +1,369 @@
+ * list the exceptions that may be propagated when the scene is rendered:
+ * </p>
+ *
+ * @exception {RuntimeError} Failed to load external reference.
+ */
+ModelExperimental.prototype.update = function (frameState) {
+  if (!defined(this._defaultTexture)) {
+    this._defaultTexture = frameState.context.defaultTexture;
+  }
+
+  // Keep processing the glTF every frame until the main resources
+  // (buffer views) and textures (which may be loaded asynchronously)
+  // are processed.
+  if (!this._resourcesLoaded || !this._texturesLoaded) {
+    this._gltfLoader.process(frameState);
+  }
+
+  // short-circuit if the glTF resources aren't ready.
+  if (!this._resourcesLoaded) {
+    return;
+  }
+
+  if (!this._drawCommandsBuilt) {
+    this._sceneGraph.buildDrawCommands(frameState);
+    this._drawCommandsBuilt = true;
+
+    var model = this;
+    // Set the model as ready after the first frame render since the user might set up events subscribed to
+    // the post render event, and the model may not be ready for those past the first frame.
+    frameState.afterRender.push(function () {
+      model._ready = true;
+      model._readyPromise.resolve(model);
+    });
+  }
+
+  if (this._debugShowBoundingVolumeDirty) {
+    updateShowBoundingVolume(this._sceneGraph, this._debugShowBoundingVolume);
+    this._debugShowBoundingVolumeDirty = false;
+  }
+
+  frameState.commandList.push.apply(
+    frameState.commandList,
+    this._sceneGraph._drawCommands
+  );
+};
+
+/**
+ * Returns true if this object was destroyed; otherwise, false.
+ * <br /><br />
+ * If this object was destroyed, it should not be used; calling any function other than
+ * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.
+ *
+ * @returns {Boolean} <code>true</code> if this object was destroyed; otherwise, <code>false</code>.
+ *
+ * @see ModelExperimental#destroy
+ */
+ModelExperimental.prototype.isDestroyed = function () {
+  return false;
+};
+
+/**
+ * Destroys the WebGL resources held by this object.  Destroying an object allows for deterministic
+ * release of WebGL resources, instead of relying on the garbage collector to destroy this object.
+ * <br /><br />
+ * Once an object is destroyed, it should not be used; calling any function other than
+ * <code>isDestroyed</code> will result in a {@link DeveloperError} exception.  Therefore,
+ * assign the return value (<code>undefined</code>) to the object as done in the example.
+ *
+ * @exception {DeveloperError} This object was destroyed, i.e., destroy() was called.
+ *
+ *
+ * @example
+ * model = model && model.destroy();
+ *
+ * @see ModelExperimental#isDestroyed
+ */
+ModelExperimental.prototype.destroy = function () {
+  var gltfLoader = this._gltfLoader;
+  if (defined(gltfLoader)) {
+    gltfLoader.destroy();
+  }
+
+  var resources = this._resources;
+  for (var i = 0; i < resources.length; i++) {
+    resources[i].destroy();
+  }
+
+  destroyObject(this);
+};
```

Comment:
Other things that need to be destroyed:

* Shaders
* Vertex arrays

These should be added to `_resources` when they are created